src/infill/LightningGenerator.h

```diff
@@ -0,0 +1,69 @@
+//Copyright (c) 2021 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#ifndef LIGHTNING_GENERATOR_H
+#define LIGHTNING_GENERATOR_H
+
+#include "LightningLayer.h"
+
+#include "../utils/polygonUtils.h"
+
+#include <functional>
+#include <memory>
+#include <vector>
+#include <vector>
+
+namespace cura 
+{
+class SliceMeshStorage;
+
+/*
+ *            .---------.
+ *     . :+*#%%@*=\\\*@@@#:
+ *  .+*=+@%#%@@@@#%@@@@@@@@*
+ * :@@%+=@##==%@@@@@@@@@@@@@
+ * %@@%=-'     '"-+*#%@@@@@@@.
+ * %@#'            ...*=@@@@@-
+ * .-             ....*=@@@@@*
+ *  .        ..:......:#@@@@@=
+ *  : :-- .-*%@%%**=-:.-%@@#=.=.
+ *   =##%: :==-.:::..:::=@+++. )
+ *   :     ..       .::--:-#% /
+ *    \    ...     ..:---==:_;
+ *     :  :=w=:   ..:----+=
+ *      :-#@@@%#*-.:::---==
+ *      :*=--==--:.:----=-=.
+ *       . .-=-...:--=+*+-+=:.
+ *        \     .-=+:'       .:
+ *         .':-==-"  .:-=+#%@@@*
+ *       .'      :+#@@@@@@@@@@@@+
+ *    .=#%#. :-+#%@@@@@@@@@@@@@@@:
+ *  -+%##*+#***%@%####%@@@@@@@@@@@*.
+ * 
+ *                           <3 Nikolai
+ */
+class LightningGenerator
+{
+public:
+    LightningGenerator(const SliceMeshStorage& mesh);
+
+    const LightningLayer& getTreesForLayer(const size_t& layer_id);
```

Comment:
I don't agree with this warning. It means your declaration will be (visually) different from the definition. It's going to trip me up a lot.

Reply:
I don't feel that strongly about this subject. But I do think static code analysis warnings are best solved. For the obvious reasons that it reduces risk but more importantly that you don't overlook other more important warnings because of the noise of these warnings.